src/worker/controller/actions/proxy/RouteRpcNonBlockingAction.js

```diff
@@ -8,24 +10,34 @@ class RouteRpcNonBlockingAction {
     const targetTopic = requestEnvelop.content().request.workerAddress;
     const request = requestEnvelop.content().request;
     const type = requestEnvelop.content().type;
+
     // validate topic indicated
     if (!targetTopic || !request) {
-      const env = new Envelop(requestEnvelop.id(), {result: {sent: false}}, requestEnvelop.type());
-      return this._controller.
-          communicator().
-          send(env);
+      this._sendResponseEnvelope(requestEnvelop, {sent: false}, 'error no targetTopic/request');
+      return;
     }
+    // encode routed msg
+    const routedMessage = EncoderUtil.encode({
+      type: type,
+      request: request,
+    });
+    if (!routedMessage) {
+      this._sendResponseEnvelope(requestEnvelop, {sent: false}, 'error in encoding routed message');
+      return;
+    }
+
     // send the request
     this._controller.execCmd(constants.NODE_NOTIFICATIONS.PUBSUB_PUB, {
       topic: targetTopic,
-      message: JSON.stringify({
-        type: type,
-        request: request,
-      }),
+      message: routedMessage,
     });
     // return jsonrpc response ack
-    const responseEnvelop = new Envelop(requestEnvelop.id(), {result: {sent: true}}, requestEnvelop.type());
-    this._controller.communicator().send(responseEnvelop);
+    this._sendResponseEnvelope(requestEnvelop, {sent: true}, null);
+  }
+
+  _sendResponseEnvelope(requestEnvelop, result, error) {
```

Comment:
is there a way to put this function in a shared library so we won't need to duplicate it?